WasatchNET/MultiChannelWrapper.cs

```diff
@@ -0,0 +1,407 @@
+        /// </summary>
+        /// <returns>-1 if none found</returns>
+        public int triggerPos => specTrigger != null ? specTrigger.multiChannelPosition : -1;
+
+        /// <summary>
+        /// Whether external hardware triggering is currently enabled.
+        /// </summary>
+        public bool triggeringEnabled
+        {
+            get => _triggeringEnabled; 
+            set
+            {
+                foreach (var pair in specByPos)
+                {
+                    logger.debug($"pos {pair.Key}: triggerEnabled -> {value}");
+                    pair.Value.triggerSource = value ? TRIGGER_SOURCE.EXTERNAL : TRIGGER_SOURCE.INTERNAL;
+                }
+                _triggeringEnabled = value;
+            }
+        }
+        bool _triggeringEnabled;
+
+        /// <summary>
+        /// The position of the spectrometer configured to control the fans.
+        /// </summary>
+        /// <returns>-1 if none found</returns>
+        public int fanPos => specFan != null ? specFan.multiChannelPosition : -1;
+
+        /// <summary>
+        /// Use the configured spectrometer's laserEnable output to turn the system fans on or off.
+        /// </summary>
+        /// <remarks>Obviously dangerous on spectrometers with physical lasers.</remarks>
+        /// <todo>move to GPIO</todo>
+        public bool fanEnabled
+        {
+            get
+            {
+                if (specFan is null)
+                {
+                    logger.error("no spectrometer configured with fan control");
+                    return false;
+                }
+                return specFan.laserEnabled;
+            }
+
+            set
+            {
+                if (specFan is null)
+                {
+                    logger.error("no spectrometer configured with fan control");
+                    return;
+                }
+                specFan.laserEnabled = value;
+            }
+        }
+
+        ////////////////////////////////////////////////////////////////////////
+        // Acquisition
+        ////////////////////////////////////////////////////////////////////////
+
+        /// <summary>
+        /// Use the configured spectrometer's laserEnable output to raise a brief 
+        /// HW trigger pulse.
+        /// </summary>
+        /// <todo>test w/ARM spectrometers</todo>
+        public async void startAcquisition()
+        {
+            if (specTrigger is null)
+            {
+                logger.error("startAcquisition: no spectrometer configured with trigger control");
+                return;
+            }
+
+            logger.debug("triggerPulse: high");
+            specTrigger.laserEnabled = true;
+
+            logger.debug("triggerPulse: wait");
+            await Task.Delay(TRIGGER_PULSE_WIDTH_MS);
+
+            logger.debug("triggerPulse: low");
+            specTrigger.laserEnabled = false;

```

Comment:
Not really sure what's going on here. Only a single spectrometer is "allowed" to have a trigger?

Reply:
Right, this is the heart of the matter.  Only one spectrometer will output a trigger signal.  When that wire/pin goes "high," every spectrometer will begin their integrations at the same time.  So technically every spectrometer is _receiving_ a trigger, but only one can generate a trigger.